test/test_profiler.py

```diff
@@ -1282,6 +1341,46 @@ def test_utils_compute_queue_depth_when_no_cuda_events(self):
         basic_evaluation = _utils.BasicEvaluation(prof.profiler)
         self.assertFalse(basic_evaluation.compute_queue_depth())
 
+    def test_utils_compute_idle_time(self):
+        profiler = self.generate_mock_profile()
+        basic_evaluation = _utils.BasicEvaluation(profiler)
+        res = ""
+        for event_key in basic_evaluation.event_keys:
```

Comment:
You can make this cleaner with a list comprehension.
```
expected = "\n".join([
    f"{basic_evaluation.metrics[event_key].idle_time_ns} [{event_key.event.name()}]"
    for event_key in basic_evaluation.event_keys])
```

It's also good not to get in the habit of appending strings in a loop, since immutable strings makes your implementation quadratic rather than linear.

Reply:
fantastic trick!